src/Tribe/RSVP.php

```diff
@@ -731,23 +1092,55 @@ public function generate_tickets() {
 
 		// If there are details missing, we return to the event page with the rsvp_error
 		if ( false === $attendee_details ) {
-			$url = get_permalink();
-			$url = add_query_arg( 'rsvp_error', 1, $url );
-			wp_redirect( esc_url_raw( $url ) );
-			tribe_exit();
+			if ( $redirect ) {
+				$url = get_permalink();
+				$url = add_query_arg( 'rsvp_error', 1, $url );
+
+				wp_redirect( esc_url_raw( $url ) );
+				tribe_exit();
+			}
+
+			return false;
+		}
+
+		$product_ids = [];
+
+		if ( isset( $_POST['tribe_tickets'] ) ) {
+			$product_ids = wp_list_pluck( $_POST['tribe_tickets'], 'ticket_id' );
+		} elseif ( isset( $_POST['product_id'] ) ) {
+			$product_ids = (array) $_POST['product_id'];
```

Comment:
* :warning: Detected usage of a non-sanitized input variable: $_POST['product_id']
* :no_entry_sign: Processing form data without nonce verification.

<sub>hash: 79a5bfcb5c8a7442dce622790ef590ef</sub>